README

```diff
@@ -8,739 +8,742 @@ This repo represents the x11vnc codebase that once resided in the LibVNCServer r
+
+You may want to add the -C option to ssh to enable compression. The
+VNC compression is not perfect, and so this may help a bit. However,
+over a fast LAN you probably don't want to enable SSH compression
+because it can slow things down. Try both and see which is faster.
+
+If your username is different on the remote machine use something
+like: "fred@far-away.east" in the above ssh command line.
+
+Some VNC viewers will do the ssh tunnelling for you automatically, the
+TightVNC Unix vncviewer does this when the "-via far-away.east" option
+is supplied to it (this requires x11vnc to be already running on
+far-away.east or having it started by inetd(8).) See the 3rd script
+example below for more info.
+
+SSVNC:  You may also want to look at the Enhanced TightVNC Viewer
+(ssvnc) bundles because they contain scripts and GUIs to automatically
+set up SSH tunnels (e.g. the GUI, "ssvnc", does it automatically and
+so does this command: "ssvnc_cmd -ssh user@far-away.east:0") and can
+even start up x11vnc as well.
+
+The Terminal Services mode of SSVNC is perhaps the easiest way to use
+x11vnc. You just need to have x11vnc available in $PATH on the remote
+side (and can SSH to the host), and then on the viewer-side you type
+something like:
+
+```
+tsvnc fred@far-away.east
+```
+
+everything else is done automatically for you. Normally this will
+start a virtual Terminal Services X session (RAM-only), but if you
+already have a real X session up on the physical hardware it will find
+that one for you.
+
+Gateways:  If the machine you SSH into is not the same machine with
+the X display you wish to view (e.g. your company provides incoming
+SSH access to a gateway machine), then you need to change the above
+to, e.g.: "-L 5900:OtherHost:5900":
+
+```
+sitting-here> ssh -t -L 5900:OtherHost:5900 gateway.east
+```
+
+Where gateway.east is the internet hostname (or IP) of the gateway
+machine (SSH server.) 'OtherHost' might be, e.g., freds-pc or
+192.168.2.33 (it is OK for these to be private hostnames or private IP
+addresses, the host in -L is relative to the remote server side.)
+
+Once logged in, you'll need to do a second login (ssh, rsh, etc.) to
+the workstation machine 'OtherHost' and then start up x11vnc on it (if
+it isn't already running.) (The "-connect gateway:59xx" option may be
+another alternative here with the viewer already in -listen mode.) For
+an automatic way to use a gateway and have all the network traffic
+encrypted (including inside the firewall) see Chaining SSH's.
+
+These gateway access modes also can be done automatically for you via
+the "Proxy/Gateway" setting in SSVNC (including the Chaining SSH's
+case, "Double Proxy".)
+
+## Chaining SSH's
+
+Note that for use of a ssh gateway and -L redirection
+to an internal host (e.g. "-L 5900:otherhost:5900") the VNC traffic
+inside the firewall is not encrypted and you have to manually log into
+otherhost to start x11vnc. Kyle Amon shows a method where you chain
```

Comment:
this is an addtition?

Reply:
Nope, this "Chaining SSH's" was moved here from line 9308 in the old README.

Original source was http://www.karlrunge.com/x11vnc/chainingssh.html